sdk/openai/openai/test/public/assisstants.spec.ts

```diff
@@ -0,0 +1,237 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { Recorder } from "@azure-tools/test-recorder";
+import { matrix } from "@azure-tools/test-utils";
+import { assert, describe, beforeEach, afterEach, it } from "vitest";
+import { assertAssistantEquality } from "./utils/asserts.js";
+import { createClient, startRecorder } from "./utils/createClient.js";
+import OpenAI, { AzureOpenAI } from "openai";
+import { APIVersion, AuthMethod, authTypes, latestAPIPreview } from "./utils/utils.js";
+import { AssistantTool } from "openai/resources/beta/assistants.mjs";
+
+interface Metadata {
+  foo: string;
+}
+describe("OpenAIAssistants", () => {
+  let recorder: Recorder;
+
+  beforeEach(async function (context) {
+    recorder = await startRecorder(context);
+  });
+
+  afterEach(async function () {
+    await recorder.stop();
+  });
+
+  matrix([authTypes] as const, async function (authMethod: AuthMethod) {
+    matrix([[latestAPIPreview]] as const, async function (apiVersion: APIVersion) {
+      describe(`[${apiVersion}] Client`, () => {
+        let client: OpenAI | AzureOpenAI;
+        const codeAssistant = {
+          tools: [{ type: "code_interpreter" }] as AssistantTool[],
```

Comment:
Is this cast needed?

Reply:
Yes, it doesn't infer the correct tool, so I can update the cast to either `tools: [{ type: "code_interpreter" as const }]` or `tools: [{ type: "code_interpreter" } as AssistantTool ]`,

Reply:
I see, let's add `as const` then.